src/Penrose/Style.hs

```diff
@@ -239,6 +245,10 @@ data Expr
     | UOp UnaryOp Expr
     | List [Expr]
     | Tuple Expr Expr
+    | Vector [Expr]
+    | Matrix [Expr]
+    | VectorAccess Expr Expr -- ^ Reference to vector, index
```

Comment:
@wodeni I just realized that the VectorAccess and MatrixAccess hardcode the dimensions of the vectors and matrices. Could you generalize them to be a list of indices (so we can deal with 3-vectors, etc.)? Thanks!